core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -86,7 +86,10 @@ private[spark] object TaskDescription {
     dataOut.writeInt(taskDescription.properties.size())
     taskDescription.properties.asScala.foreach { case (key, value) =>
       dataOut.writeUTF(key)
-      dataOut.writeUTF(value)
+      // SPARK-19796 -- writeUTF doesn't work for long strings, which can happen for property values
+      val bytes = value.getBytes("utf-8")
+      dataOut.writeInt(bytes.length)
+      dataOut.write(bytes)
```

Comment:
Oops realized I was looking at ObjectOutputStream and we use DataOutputStream here, so my proposal is no good.  This is fine.  Also I checked how this affects the serialized size and this only increases it by 6 bytes relative to writeUTF.